client/src/components/CarsFilter/index.tsx

```diff
@@ -44,44 +46,93 @@ const getYears = ():Array<string> => {
-  const [fuel, setFuel] = useState('');
-  const [maxPrice, setMaxPrice] = useState(0);
-  const [isGoodPrice, setPriceBool] = useState(false);
+  const [brand, setBrand] = useState<string>(state?.brand || '');
+  const [model, setModel] = useState<string | null>('');
+  const [mileage, setMileage] = useState<number | number[] >(0);
+  const [year, setYear] = useState<string | null>(null);
+  const [fuel, setFuel] = useState<string | null>('');
+  const [maxPrice, setMaxPrice] = useState<number>(0);
+  const [isGoodPrice, setPriceBool] = useState<boolean>(false);
 
   const changePriceType = (event: React.ChangeEvent<HTMLInputElement>) => {
     setPriceBool(event.target.checked);
   };
 
-  const changeBrand = (e: any, values: SetStateAction<any>) => {
-    setBrand(values);
+  const changeBrand = (
+    event: React.SyntheticEvent<Element, Event>,
+    value: SetStateAction<string>,
+  ) => {
+    setBrand(value);
   };
-  const changeModel = (e: any, values: SetStateAction<any>) => {
-    setModel(values);
+  const changeModel = (
+    event: React.SyntheticEvent<Element, Event>,
+    value: string | null,
+  ) => {
+    setModel(value);
   };
-  const changeYear = (e: any, values: SetStateAction<any>) => {
-    setYear(values);
+  const changeYear = (
+    event: React.SyntheticEvent<Element, Event>,
+    value: string | null,
+  ) => {
+    setYear(value);
   };
-  const changeMileage = (e: any, values: SetStateAction<any>) => {
-    setMileage(values * 10000);
+  const changeMileage = (
+    event: Event,
+    value: number | number[],
+  ) => {
+    setMileage(+value * 10000);
   };
-  const changefuelType = (e: React.SyntheticEvent<Element, Event>, values: SetStateAction<any>) => {
-    setFuel(values);
+  const changefuelType = (
+    event: React.SyntheticEvent<Element, Event>,
+    value: SetStateAction<string | null>,
+  ) => {
+    setFuel(value);
   };
   const changeMaxPrice = (event: React.ChangeEvent<HTMLInputElement>) => {
     const num = Number(event.target.value);
     setMaxPrice(num);
   };
 
   useEffect(() => {
-    // here we will fetch the data and set it on a props that it defined in parent component
-  }, [brand, model, mileage, year, fuel, isGoodPrice]);
+    setLoading(true);
+    const params:Params = {
+      page: pageNumber,
+    };
+    if (model?.length !== 0 || search.length !== 0) {
+      params.model = model || search;
+    }
+    if (brand?.length !== 0) {
+      params.brand = brand;
+    }
+    if (year?.length !== 0) {
+      params.year = year;
+    }
+    if (fuel?.length !== 0) {
+      params.fuel = fuel;
+    }
+    if (maxPrice > 0) {
+      params.maxPrice = maxPrice;
+    }
+    if (isGoodPrice) {
+      params.goodPrice = 1;
+    }
+    if (mileage !== 0) {
+      params.mileage = mileage;
+    }
+    const getCars = async () => {
+      // eslint-disable-next-line max-len
+      const response: CarsData = await httpInstance.get('/cars?', { params });
```

Comment:
What if there's an error!
create try and catch block and create an error state if there's an error, maybe we can use a snackbar component to tell the user that there's unexpected error